app/concerns/koudoku/subscription.rb

```diff
@@ -189,15 +224,20 @@ def subscription_owner_email
   end
 
   def changing_plans?
-    plan_id_changed?
+    will_save_change_to_plan_id?
   end
 
   def downgrading?
-    plan.present? and plan_id_was.present? and plan_id_was > self.plan_id
+    plan.present? and plan_id_before_last_save.present? and plan_id_before_last_save > self.plan_id
   end
 
   def upgrading?
-    (plan_id_was.present? and plan_id_was < plan_id) or plan_id_was.nil?
+    (plan_id_before_last_save.present? and plan_id_before_last_save < plan_id) or plan_id_before_last_save.nil?
+  end
+
+  # CF Template methods.
+  def subscription_options(opts = {})
```

Comment:
This is the relevant new method. I'm not sure what the other stuff is or why it's showing up in this PR. I guess those changes haven't been merged to master? Or maybe they were reverted?